### PR TITLE
Add PLATO to Catalogs

### PR DESCRIPTION
**Pull request by jaymedina**
*Functionality*: Following the latest MAST patch, this PR adds PLATO as a query option to `astroquery.mast.Catalogs.query_region` and `query_object`.

*Documentation*: `mast.rst` is updated to reflect new PLATO DR1 availability. 

*Testing*: `test_mast_remote.py` is updated with a regression test for `PLATO` query.

### CODE REVIEW

**Comment by jaymedina** [comment]
Latest check will pass once the latest MAST patch gets deployed. I think by end of today or tomorrow.

File: CHANGES.rst [hunk 1]
```diff
@@ -29,6 +29,8 @@ mast
 
 - Adding moving target functionality to ``astroquery.mast.Tesscut`` [#2121]
 
+- Adding PLATO catalog as query option in ``astroquery.mast.Catalogs.query_object`` and ``astroquery.mast.Catalogs.query_region``. [#2279]
```
Comment (bsipocz) [inline CHANGES.rst:32]: This is the PLATO input catalogue, right? I would also rephrase a little to concise the entry, but happy to do that before release when I do a changelog cleanup
```suggestion
- Adding the All-Sky PLATO Input Catalog ('plato') as a catalog option for methods of ``astroquery.mast.Catalogs``. [#2279]
```

File: astroquery/mast/collections.py [hunk 2]
```diff
@@ -42,7 +42,9 @@ def __init__(self):
 
         services = {"panstarrs": {"path": "panstarrs/{data_release}/{table}.json",
                                   "args": {"data_release": "dr2", "table": "mean"}}}
-        self._service_api_connection.set_service_params(services, "catalogs", True)
+
+        self._service_api_connection.set_service_params(services, "catalogs",
+        True)
```
Comment (bsipocz) [inline astroquery/mast/collections.py:47]: revert this formatting change. I think you could avoid adding them in the future by adding a less conservative line length into your editor (e.g. 100 or 120 could work, you can always break long lines manually where they make logical sense, and then avoid reformats like this where the new one is neither more legible nor strictly following pep8 conventions)

File: astroquery/mast/collections.py [hunk 3]
```diff
@@ -141,10 +143,17 @@ def query_region_async(self, coordinates, radius=0.2*u.deg, catalog="Hsc",
                 if version == 1:
                     service = "Mast.Catalogs.GaiaDR1.Cone"
                 else:
-                    if version not in (2, None):
+                    if version not in (None, 2):
```
Comment (bsipocz) [inline astroquery/mast/collections.py:146]: why this change?

File: docs/mast/mast.rst [hunk 4]
```diff
@@ -458,7 +458,8 @@ The Catalogs class provides access to a subset of the astronomical catalogs stor
 - The TESS Input Catalog (TIC)
 - The TESS Candidate Target List (CTL)
 - The Disk Detective Catalog
-- PanSTARRS (DR1, DR2)
+- The PanSTARRS Catalog (DR1 and DR2)
+- The PLATO Catalog (DR1)
```
Comment (bsipocz) [inline docs/mast/mast.rst:462]: Is this the asPIC catalog or something else? Either case, calling it PLATO catalog DR1 is confusing given PLATO hasn't yet been launched.

**Review by bsipocz — COMMENTED** [review]
Minor comments only. Please squash commits though.

File: astroquery/mast/collections.py (hunk 3 above)
Comment (jaymedina) [reply astroquery/mast/collections.py:146]: it made more sense to me to have `None` go first

File: docs/mast/mast.rst (hunk 4 above)
Comment (jaymedina) [reply docs/mast/mast.rst]: Yes, this is referring to asPIC, and I just confirmed w a team member. Documentation is now updated to reflect this

File: astroquery/mast/collections.py [hunk 5]
```diff
@@ -43,7 +43,7 @@ def __init__(self):
         services = {"panstarrs": {"path": "panstarrs/{data_release}/{table}.json",
                                   "args": {"data_release": "dr2", "table": "mean"}},
                     "plato": {"path": "plato/{data_release}/{table}.json",
-                                         "args": {"data_release": "dr2", "table": "mean"}}}
+                                         "args": {"data_release": ["dr1", "dr2"], "table": "mean"}}}
```
Comment (bsipocz) [inline astroquery/mast/collections.py]: Shouldn't this be panstarrs?

Comment (jaymedina) [reply astroquery/mast/collections.py]: Hmm I'm not sure how that snuck in again, I thought I'd removed it. Yes this shouldn't be here at all.

Comment (jaymedina) [reply astroquery/mast/collections.py]: I think this is from an older commit because I don't see it in my local branch and I also don't see it under `Files Changed`

File: astroquery/mast/collections.py [hunk 6]
```diff
@@ -42,6 +42,7 @@ def __init__(self):
 
         services = {"panstarrs": {"path": "panstarrs/{data_release}/{table}.json",
                                   "args": {"data_release": "dr2", "table": "mean"}}}
+
```
Comment (bsipocz) [inline astroquery/mast/collections.py:45]: Now I'm confused: at some iterations plato was added to this one, too, but then I don't see it's being used anywhere. So what is this attribute used for, and why is `panstarrs` the only element in there?

Comment (jaymedina) [reply astroquery/mast/collections.py:45]: When I first started this PR, I added `plato` to the services call before realizing I didn't need to because plato was in the Mashup API already, so I removed it from this services call a few commits later. I think what's happening is if you're looking at individual commits instead of `Files Changed`, you're seeing a version of this branch where I had `plato` added, but I removed it later down the line.

`panstarrs` has its own special `services` call because `panstarrs` is not part of the Mashup API. It's only part of the Catalogs service. So that's why you'll see a bunch of `if, else` statements where, if the catalog is indeed `panstarrs`, the database gets called from another location using `service_api_connection` instead of `portal_api_connection`. See example here: Lines 111 to 115 https://github.com/astropy/astroquery/blob/5740dc772a1924e3446b85bbf898d4f9f8fc3ea7/astroquery/mast/collections.py#L111

Comment (bsipocz) [reply astroquery/mast/collections.py:45]: > panstarrs has its own special services call because panstarrs is not part of the Mashup API.

OK, fine, this is the info I was after, that panstarrs is actually the odd one out.

Comment (bsipocz) [reply astroquery/mast/collections.py:45]: btw, I made this comment on the overall diff rather than on the commit level, yet it brings it up as I had made it on the first commit. So I indeed think it would be useful to squash all together to remove that temporary change in the middle. After all we occasionally go back to the commit level.

Comment (jaymedina) [reply astroquery/mast/collections.py:45]: Sure thing!
